employees/tests/test_unit_report_viewset.py

```diff
@@ -417,6 +490,44 @@ def test_custom_report_detail_view_should_not_update_report_on_post_if_form_is_i
         self.assertIsNotNone(response.data['errors'])
         self.assertNotEqual(new_description, self.report.description)
 
+    def test_custom_report_detail_view_should_not_update_report_if_author_is_not_a_member_of_selected_project(self):
+        other_project = Project(
+            name="Other Project",
+            start_date=datetime.datetime.now(),
+        )
+        other_project.full_clean()
+        other_project.save()
+        new_description = 'Some other description'
+        request = APIRequestFactory().post(
+            path=reverse('custom-report-detail', args=(self.report.pk,)),
+            data={
+                'date': datetime.datetime.now().date(),
+                'description': new_description,
+                'project': other_project,
+                'work_hours': Decimal('8.00'),
+            },
+        )
+        request.user = self.user
+        response = ReportDetail.as_view()(request, pk=self.report.pk)
+        self.report.refresh_from_db()
+        self.assertEqual(response.status_code, 200)
+        self.assertIsNotNone(response.data['errors'])
```

Comment:
Please change this assert that you check exactly what error is in this dict

Reply:
In order to get to the ErrorDetail object, I have to do this:
`response.data['errors']['project'][0]`

Then I can either get to the code
`response.data['errors']['project'][0].code = 'does_not_exist'`
or to the (very specific) message
`str(response.data['errors']['project'][0]) = 'Object with name=Other Project does not exist.'`

Which one should I used? Or should just narrow it down to checking if field specific errors exist? `self.assertIsNotNone(response.data['errors']['project'])`

Reply:
@maciejSamerdak I would prefer to check `response.data['errors']['project'][0].code = 'does_not_exist'` but I know that to change it everywhere is very big effort, so please just apply this to this PR.

Reply:
Done